libtest/signal.cc

```diff
@@ -225,13 +240,15 @@ SignalThread::SignalThread() :
 bool SignalThread::setup()
 {
   set_shutdown(SHUTDOWN_RUNNING);
-  lock = sem_open(lock_name, O_CREAT|O_EXCL);
+
+  const char * lock_name = random_lock_name().c_str();
+  lock = sem_open(lock_name, O_CREAT|O_EXCL, 0660, 0);
```

Comment:
Instead of `0660`, use `S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP` instead?

I think those `S_*` constants are defined in `sys/stat.h`?

```C
#include <sys/stat.h>
```

Reply:
OK